src/idl_gen_general.cpp

```diff
@@ -587,6 +587,19 @@ std::string GenGetter(const Type &type) {
   }
 }
 
+// Returns the function name that is able to read a value of the given type.
+std::string GenGetterForLookupByKey(const Type &type,
+                                    const std::string data_buffer) {
+  auto getter = data_buffer + "." + FunctionStart('G') + "et";
```

Comment:
part of this code is copy pasted from the function just above it.. please share that code in a sub-function.